core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLRecordReader.java

```diff
@@ -0,0 +1,305 @@
+/*
+  * Copyright © 2016 Cask Data, Inc.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+  * use this file except in compliance with the License. You may obtain a copy of
+  * the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+  * License for the specific language governing permissions and limitations under
+  * the License.
+  */
+
+package co.cask.hydrator.plugin.batch.source;
+
+import co.cask.hydrator.plugin.common.FileUtililty;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Map;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+/**
+ * XMLRecordReader class to read through a given xml document and to output xml blocks as per node path specified.
+ */
+public class XMLRecordReader extends RecordReader<LongWritable, Map<String, String>> {
+  private static final Logger LOG = LoggerFactory.getLogger(XMLRecordReader.class);
+
+  public static final String CLOSING_END_TAG_DELIMITER = ">";
+  public static final String OPENING_END_TAG_DELIMITER = "</";
+  public static final String CLOSING_START_TAG_DELIMITER = ">";
+  public static final String OPENING_START_TAG_DELIMITER = "<";
+
+  private LongWritable currentKey;
+  private Map<String, String> currentValue;
+  private String nodePath;
+  private final String fileName;
+  private final XMLStreamReader reader;
+  private final String[] nodes;
+  private final int totalNodes;
```

Comment:
do you need to store this as a member? This is the same as the size of the maps below right?